common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -4,50 +4,267 @@
+        enforce_type=FEATURES.get('ENABLE_XBLOCK_XML_VALIDATION', True),
+        default="<problem></problem>"
+    )
+    correct_map = Dict(help=_("Dictionary with the correctness of current student answers"),
+                       scope=Scope.user_state, default={})
+    input_state = Dict(help=_("Dictionary for maintaining the state of inputtypes"), scope=Scope.user_state)
+    student_answers = Dict(help=_("Dictionary with the current student responses"), scope=Scope.user_state)
+
+    # enforce_type is set to False here because this field is saved as a dict in the database.
+    score = ScoreField(help=_("Dictionary with the current student score"), scope=Scope.user_state, enforce_type=False)
+    has_saved_answers = Boolean(help=_("Whether or not the answers have been saved since last submit"),
+                                scope=Scope.user_state, default=False)
+    done = Boolean(help=_("Whether the student has answered the problem"), scope=Scope.user_state, default=False)
+    seed = Integer(help=_("Random seed for this student"), scope=Scope.user_state)
+    last_submission_time = Date(help=_("Last submission time"), scope=Scope.user_state)
+    submission_wait_seconds = Integer(
+        display_name=_("Timer Between Attempts"),
+        help=_("Seconds a student must wait between submissions for a problem with multiple attempts."),
+        scope=Scope.settings,
+        default=0)
+    weight = Float(
+        display_name=_("Problem Weight"),
+        help=_("Defines the number of points each problem is worth. "
+               "If the value is not set, each response field in the problem is worth one point."),
+        values={"min": 0, "step": .1},
+        scope=Scope.settings
+    )
+    markdown = String(help=_("Markdown source of this module"), default=None, scope=Scope.settings)
+    source_code = String(
+        help=_("Source code for LaTeX and Word problems. This feature is not well-supported."),
+        scope=Scope.settings
+    )
+    use_latex_compiler = Boolean(
+        help=_("Enable LaTeX templates?"),
+        default=False,
+        scope=Scope.settings
+    )
+    matlab_api_key = String(
+        display_name=_("Matlab API key"),
+        help=_("Enter the API key provided by MathWorks for accessing the MATLAB Hosted Service. "
+               "This key is granted for exclusive use by this course for the specified duration. "
+               "Please do not share the API key with other courses and notify MathWorks immediately "
+               "if you believe the key is exposed or compromised. To obtain a key for your course, "
+               "or to report an issue, please contact moocsupport@mathworks.com"),
+        scope=Scope.settings
+    )
+
+    INDEX_CONTENT_TYPE = 'CAPA'
+
+    resources_dir = None
+
+    has_score = True
+    show_in_read_only_mode = True
+    template_dir_name = 'problem'
+    mako_template = "widgets/problem-edit.html"
+    has_author_view = True
```

Comment:
The attributes were moved up from their location on the top of the `CapaDescriptor` below without any changes.